[CI] Problems when running tests in Windows Latest GHA runner for 3.13

https://github.com/pypa/setuptools/actions/runs/15653392162/job/44101150498#step:11:127


```
>       assert code == 0, f'\nSTDOUT:\n{stdout}\nSTDERR:\n{stderr}'
E       AssertionError: 
E         STDOUT:
E         running build
E         running build_ext
E         building 'foo' extension
E         creating foo_build\temp.win-amd64-cpython-313\Release
E         "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -ID:\a\setuptools\setuptools\.tox\py\include -IC:\hostedtoolcache\windows\Python\3.13.4\x64\include -IC:\hostedtoolcache\windows\Python\3.13.4\x64\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.26100.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tcfoo.c /Fofoo_build\temp.win-amd64-cpython-313\Release\foo.obj
E         foo.c
E         creating C:\Users\runneradmin\AppData\Local\Temp\pytest-of-unknown\pytest-0\popen-gw2\test_build_ext_config_handling0\foo_build\lib.win-amd64-cpython-313
E         "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\bin\HostX86\x64\link.exe" /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:D:\a\setuptools\setuptools\.tox\py\libs /LIBPATH:C:\hostedtoolcache\windows\Python\3.13.4\x64\libs /LIBPATH:C:\hostedtoolcache\windows\Python\3.13.4\x64 /LIBPATH:D:\a\setuptools\setuptools\.tox\py\PCbuild\amd64 "/LIBPATH:C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\ATLMFC\lib\x64" "/LIBPATH:C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.26100.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\\lib\10.0.26100.0\\um\x64" /EXPORT:PyInit_foo foo_build\temp.win-amd64-cpython-313\Release\foo.obj /OUT:foo_build\lib.win-amd64-cpython-313\foo.cp313-win_amd64.pyd /IMPLIB:foo_build\temp.win-amd64-cpython-313\Release\foo.cp313-win_amd64.lib
E         LINK : fatal error LNK1104: cannot open file 'python313t.lib'
E         
E         STDERR:
E         error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.43.34808\\bin\\HostX86\\x64\\link.exe' failed with exit code 1104
E         
E       assert 1 == 0

D:\a\setuptools\setuptools\setuptools\tests\test_build_ext.py:293: AssertionError
```


Possibly a change in environment set since the last time it run successfully?


```console
Current runner version: '2.325.0'
Operating System
  Microsoft Windows Server 2022
  10.0.20348
  Datacenter
Runner Image
  Image: windows-2022
  Version: 20250609.2.0
  Included Software: https://github.com/actions/runner-images/blob/win22/20250609.2/images/windows/Windows2022-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/win22%2F20250609.2
Runner Image Provisioner
  2.0.437.1

---

Run actions/setup-python@v5
  with:
    python-version: 3.13
    allow-prereleases: true
    check-latest: false
    token: ***
    update-environment: true
    freethreaded: false
  env:
    FORCE_COLOR: 1
    PIP_DISABLE_PIP_VERSION_CHECK: true
    PIP_NO_WARN_SCRIPT_LOCATION: true
    TOX_OVERRIDE: testenv.pass_env+=GITHUB_*,FORCE_COLOR
    SETUPTOOLS_USE_DISTUTILS: local
Installed versions
  Successfully set up CPython (3.13.4)
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -42,7 +42,7 @@ jobs:
       matrix:
         python:
         - "3.9"
-        - ">=3.13.5"
+        - ">=3.13.5"  # temporary bound until it becomes the default, python/cpython#135151
         platform:
         - ubuntu-latest
         - macos-latest

